pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -3532,6 +3532,12 @@ private CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final TopicName
             return CompletableFuture.completedFuture(true);
         }
 
+        // If this topic name contains "-partition--x" , not allow to be created automatically.
```

Comment:
This would be a more accurate description:
```suggestion
        // Skip auto-creation for topics with "-partition-" suffix missing a partition number
```